lib/writeUris.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+var fsExtra = require('fs-extra');
+var mime = require('mime');
+var path = require('path');
+var Promise = require('bluebird');
+
+var fsExtraOutputFile = Promise.promisify(fsExtra.outputFile);
+
+module.exports = writeUris;
+
+/**
+ * Write glTF resources as embedded data uris or external files.
+ *
+ * @param {Object} gltf The glTF asset.
+ * @param {String} gltfPath Path where the glTF will be saved.
+ * @param {Boolean} separateBuffers Writes out separate buffers.
+ * @param {Boolean} separateTextures Writes out separate textures.
+ * @returns {Promise} A promise that resolves to the glTF asset.
+ *
+ * @private
+ */
+function writeUris(gltf, gltfPath, separateBuffers, separateTextures) {
+    var promises = [];
+
+    var buffer = gltf.buffers[Object.keys(gltf.buffers)[0]];
+    var bufferByteLength = buffer.extras._obj2gltf.source.length;
+
+    var texturesByteLength = 0;
+    var images = gltf.images;
+    for (var id in images) {
+        if (images.hasOwnProperty(id)) {
+            texturesByteLength += images[id].extras._obj2gltf.source.length;
+        }
+    }
+
+    // Buffers larger than ~192MB cannot be base64 encoded due to a NodeJS limitation. Source: https://github.com/nodejs/node/issues/4266
+    var exceedsMaximum = (texturesByteLength + bufferByteLength > 201326580);
```

Comment:
We should write up an obj2gltf issue for this.  There's no reason we can't use a third party module or write our own simple base64 encode function to avoid any "all at once" limitation that Node has.